earthmover/nodes/destination.py

```diff
@@ -132,3 +136,79 @@ def render_row(self, row: pd.Series):
             raise
 
         return json_string
+
+
+class CsvDestination(Destination):
+    """
+
+    """
+    mode: str = 'csv'  # Documents which class was chosen.
+    allowed_configs: Tuple[str] = (
+        'debug', 'expect', 'show_progress', 'repartition', 'source',
+        'extension', 'header', 'separator', 'limit', 'keep_columns'
+    )
+
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.header = self.error_handler.assert_get_key(self.config, 'header', dtype=bool, required=False, default=True)
+        self.separator = self.error_handler.assert_get_key(self.config, 'separator', dtype=str, required=False, default=",")
+        self.limit = self.error_handler.assert_get_key(self.config, 'limit', dtype=int, required=False, default=None)
+        self.extension = self.error_handler.assert_get_key(self.config, 'extension', dtype=str, required=False, default="csv")
```

Comment:
This field is technically required, since it has to be populated to initialize the CSV Destination.

Reply:
I used a different approach [here](https://github.com/edanalytics/earthmover/pull/89/files#r1600195842), adding another `kind` property which is used to select the destination Class. I envision that `kind` could have values like
```
file.jsonl
file.csv
file.tsv
file.parquet
file.xml
...
database.mysql
database.postgres
database.snowflake
...
```
This does perhaps make `extension` superfluous, except that if someone insists on using `.ndjson` for JSONL, or even really wants to put TSV data in a file with a `.xml` extension, I guess that should be possible. Maybe eventually `extension` becomes optional, with a default value of "infer."

Eventually we may add an (optional) `location` indicating where to materialize files or execute `database.*` SQL, with values like
```
local # (default)
s3://bucket_name/path/to/dir/
sftp://user:pass@domain.com/path/to/dir/
postgres://user:pass@domain.com:123/database_name?currentSchema=schema_name # (a SQLalchemy connection string)
snowflake://username:password@account_id/db_name/schema_name?warehouse=wh_name&role=role_name # (a SQLalchemy connection string)
...
```
(earthmover could parse the `location` and figure out what connector/library to use internally.)

`extension` also doesn't really make sense for `database.*` destination `kind`s, unless `location` is a file system in which case it would probably be `.sql`.

Eventually we may add an optional `mode: overwrite # or append`. For file materialization, this is self-explanatory. For `database.*` materialization, this could trigger a `TRUNCATE` statement before `INSERTS` begin.

One other relatively unrelated comment, if/when we support writing to databases, the _order_ in which we process destinations will become important (if there are primary/foreign key references in the data). Currently there's no way in earthmover to control the order in which destinations are processed, we'd have to figure out how to handle that... maybe (like dbt does) an optional `depends_on` property in each destination.